rand_core/src/lib.rs

```diff
@@ -185,22 +207,20 @@ pub trait SeedableRng<Seed>: Rng {
     fn from_seed(seed: Seed) -> Self;
 }
 
-
-/// Error type for cryptographic generators. Technically external generators
-/// such as the operating system or hardware generators could fail. A PRNG
-/// (algorithm) could also fail if it detects cycles, though most PRNGs have
-/// sufficiently long cycles that looping is not usually feasible.
-/// 
-/// TODO: how should error details be reported?
-#[derive(Debug)]
-pub struct Error;
-
-#[cfg(feature="std")]
-impl From<::std::io::Error> for Error {
-    fn from(_: ::std::io::Error) -> Error {
-        Error
-    }
+/// Error type for random number generators.
+#[derive(Copy, PartialEq, Eq, Clone, Debug)]
+pub enum RngError {
+    /// This RNG is permanently unavailable.
+    NotAvailable,
+    /// Needs (re)seeding. Depending on the RNG this can mean you have to wait
+    /// and retry later, or you have to (re)seed it.
+    EntropyLow,
+    /// Failure, retry for a limited number of times.
+    TemporaryFailure,
+    /// Interrupted, usually by a signal from the OS.
+    Interrupted,
+    /// Any RNG error not part of this list.
+    Other,
+    #[doc(hidden)]
+    __Nonexhaustive,
 }
-
-/// Result type (convenience type-def)
-pub type Result<T> = ::std::result::Result<T, Error>;
```

Comment:
Why remove this?

Reply:
Have you seen my reason in the PR description?

In part the reason is that I was bitten by the redefined Result type ;-).
But I don't really see a reason to have it, as we only use it in the trait definitions in a handful of places.

In a library that uses the `std::result::Result`, would users also import `rand::Result`? Or would they want to use the std one and import our `rand::Error`?

It seems to me (but I am inexperienced) browsing documentation and copying examples is easier without the redefine...